src/main/scala/no/ndla/draftapi/service/WriteService.scala

```diff
@@ -315,5 +318,34 @@ trait WriteService {
         .flatMap(id => conceptRepository.newEmptyConcept(id, externalIds))
     }
 
+    def storeFile(file: FileItem): Try[api.UploadedFile] =
+      uploadFile(file).map(f => api.UploadedFile(f.fileName, f.contentType, f.fileExtension, s"/files/${f.filePath}"))
+
+    private[service] def getFileExtension(fileName: String): (String, Option[String]) =
+      fileName.lastIndexOf(".") match {
+        case index: Int if index > -1 => (fileName.substring(0, index), Some(fileName.substring(index)))
+        case _                        => (fileName, None)
+      }
+
+    private[service] def uploadFile(file: FileItem): Try[domain.UploadedFile] = {
+      val (filenameWithoutExt, fileExtension) = getFileExtension(file.name)
+
+      val contentType = file.getContentType.getOrElse("")
+      val fileName = Stream
+        .continually(fileNameWithRandomElement(filenameWithoutExt, fileExtension.getOrElse("")))
+        .dropWhile(fileStorage.resourceExists)
+        .head
+
+      fileStorage
+        .uploadResourceFromStream(new ByteArrayInputStream(file.get), fileName, contentType, file.size)
+        .map(uploadPath => domain.UploadedFile(fileName, uploadPath, file.size, contentType, fileExtension))
+    }
+
+    private[service] def fileNameWithRandomElement(filename: String, extension: String, length: Int = 20): String = {
+      val extensionWithDot =
+        if (!extension.headOption.contains('.') && extension.length > 0) s".$extension" else extension
+      val randomString = Random.alphanumeric.take(max(length - extensionWithDot.length, 1)).mkString
+      s"$filename-$randomString$extensionWithDot"
```

Comment:
Liker veldig godt at vi randomizer fil-navnet! :) Men jeg lurer på om vi ikke bare skal droppe hele `filename` som kommer fra brukerside.. Føles litt tryggere hvis vi kontrollerer navnet fullt ut, og vi åpner ikke opp for potensielle sikkerhetssvakheter.
Ideelt sett bør `extension` også vaskes. Eventuelt via en whitelist over godkjente filtyper.